src/gallium/drivers/lima/ir/pp/lower.c

```diff
@@ -234,6 +234,36 @@ static bool ppir_lower_vec_to_scalar(ppir_block *block, ppir_node *node)
    return true;
 }
 
+static bool ppir_lower_texture(ppir_block *block, ppir_node *node)
+{
+   ppir_node *pred;
+
+   /* We expect a single predecessor for tex load node */
+   if (!ppir_node_has_single_pred(node)) {
+      ppir_debug("ldtex has several preds\n");
+      return false;
+   }
+
+   pred = ppir_node_first_pred(node);
+   if (pred->type == ppir_node_type_load &&
+       pred->op == ppir_op_load_varying) {
+      /* If ldtex is the only successor of load_varying node
+       * we're good. Just change load_varying op type to load_coords.
+       */
+      if (ppir_node_has_single_succ(pred)) {
+         pred->op = ppir_op_load_coords;
+         ppir_load_node *load = ppir_node_to_load(pred);
+         load->dest.type = ppir_target_pipeline;
```

Comment:
We'd better keep change dest to pipeline reg to the node_to_instr stage.

Reply:
OK